### DISCUSSION TITLE
Using the Prometheus exporter

### DISCUSSION THREAD

**Discussion by aftix** [open-telemetry/opentelemetry-go#3537] [Q&A] [answered]
I can not seem to figure out how to use the prometheus exporter to send telemetry data to Grafana. I don't know how to set the endpoint URL, or how to get a trace.SpanExporter out of a prometheus *Exporter as I seem to need to (to get a TracerProvider).

**Comment by dmathieu** [comment, marked as answer]
You seem to be mixing traces and metrics. Prometheus is a metrics system, it can only expose that factor, not traces.
Did you look at the prometheus example? It shows how to configure it in a sample app, and exposes a few metrics: https://github.com/open-telemetry/opentelemetry-go/blob/main/example/prometheus/main.go

However, with prometheus, you won't "send" data, as it's a pull-based system. The prometheus exporter will allow you to expose a `/metrics` endpoint that prometheus can parse. Then, grafana can hook into your prometheus instance and use the data from there.

**Comment by aftix** [comment]
Yes, I confused metrics and traces. I see now that the Jaeger exporter will be more appropriate for me.
